06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.index = {};
+    this.object = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+  }
+  /**
+   * Gets Indexes
+   * @param {jsonObj} object receives json object
+   * @param {string} fileName of the books
+   * @return {object} returns an object of Indexes
+   */
+  createIndex(object, fileName) {
+    Object.keys(object).forEach((titles) => {
+      this.allFilesTitle.push(object[titles].title);
+    });
+    this.allLength[fileName] = object.length;
+    object.forEach((document, position) => {
+      const words = document.text.toLowerCase().match(/\w+/g);
+      words.forEach((word) => {
+        if (this.index[word]) {
+          if (!this.index[word][position]) {
+            this.index[word][position] = true;
+          }
+        } else {
+          const oneIndex = {};
+          oneIndex[position] = true;
+          this.index[word] = oneIndex;
+        }
+      });
+    });
+    this.allFiles[fileName] = this.index;
+    this.index = {};
+    return this.allFiles[fileName];
+  }
+  /**
+   * Search function
+   * @param {string} queryString word to Search
+   * @param {fileName} filterName book name to search
```

Comment:
Is there a datatype called `fileName` now?

Also, I'm not sure I understand what `filterName` is. #DescriptiveVariableNames